_scripts/customJS/tokenParser.js

```diff
@@ -0,0 +1,575 @@
+
+        return token
+
+        // complicated regex kept as a comment for reference //
+
+        // this is a complicated regex, but basically it is 6 groups:
+        //      first, prefix: contained in (), any letters or spaces
+        //      second, token: letters only
+        //      third, filter: digit-digit, plus filter options
+        //      fourth, first item format string: the standard format string options. Can be separated from the filter with an optional semi-colon
+        //      fifth, the all item format string: the standard format string options. Must be separated from the filter with a semi colon
+        //      sixth, suffix: contained in (), any letters or spaces
+        // const regexp = /<(\([()a-zA-Z-:*\s]+\))?([a-zA-Z]+):?([0-9]*[-]?[0-9]*[RrPpLlIiOoFf!]*)?;?([UutsaAnyxqQ]+)?;?([UutsaAnyxqQ]+)?(\([()a-zA-Z-:*\s]+\))?>/g; 
+        // 
+        // 
+        // if (((input || '').match(regexp) || []).length !== 1) {
+        //    return null;
+        // }
+        //
+        // let match = regexp.exec(input)
+        // if (match) {
+        //    token.prefix = match[1] ?? ""
+        //    token.token = match[2] ?? ""
+        //    token.filter = match[3] ?? ""
+        //    token.format = match[4] ?? ""
+        //    token.chainFormat = match[5] ?? match[4] ?? ""
+        //    token.suffix = match[6] ?? ""
+        //}
+
+    }
+
+    #getParameterCaseInsensitive(object, key) {
+        // get case insensitive parameter from object
+        const asLowercase = key.toLowerCase();
+        return object[Object.keys(object)
+            .find(k => k.toLowerCase() === asLowercase)
+        ];
+    }
+
+    #getFormattedCaseString(value, token) {
+        // this is something that can never be meaningfully linked
+        // and never has an article
+        // but which might need casing rules
+
+        const { NameManager } = customJS;
+
+        // this probably just returns itself, but...
+        let name = { name: value, linkText: "", indefiniteArticle: "", definiteArticle: "", linkTarget: undefined }
+
+        return NameManager.formatName(name, token.format)
+    }
+
+    #getFormattedName(value, token, metadata) {
+        // returns a formatted name
+        // pass the format and the name to the name manager to generate a formatted name
+
+        const { NameManager } = customJS;
+        let name = undefined
+
+        // check to see if we have a name object
+        if (value?.isNormalizedName) {
+            // we have a name object
+            // this probably just returns itself, but...
+            name = NameManager.getNameObject(value)
+        } else {
+            // see if we can build a name object
+            let possibleAliasKey = token.token + "Alias"
+            let alias = this.#getParameterCaseInsensitive(metadata, possibleAliasKey)
+            let sourcePageType = NameManager.getPageType(metadata)
+            if (alias) {
```

Comment:
You don’t need to check alias. It’s fine to pass the override with alias pointing to undefined.